### PR TITLE
Implementing a Write Operation With an Different Output

### PR DESCRIPTION
**Pull request by Enz000**
Hello 
I didn't found the documentation about how to return a different output for a Write operation

### CODE REVIEW

**Comment by yivi** [comment]
It would be great if there were examples for all the doc formats (Attributes as shown, but also YAML and XML).

File: core/dto.md [hunk 1]
```diff
@@ -119,3 +119,47 @@ final class BookRepresentationProvider implements ProviderInterface
     }
 }
 ```
+
+## Implementing a Write Operation With an Output Different From the Resource
+
+The logic stay almost the same than for a [State Provider](./state-providers.md).
```
Comment (alanpoulain) [inline core/dto.md:125]:
```suggestion
The logic stays almost the same than for a [State Provider](./state-providers.md).
```
I don't think this sentence is necessary though.

File: core/dto.md [hunk 2]
```diff
@@ -119,3 +119,47 @@ final class BookRepresentationProvider implements ProviderInterface
...
+namespace App\Entity;
+
+use ApiPlatform\Metadata\Post;
+use App\Dto\AnotherRepresentation;
+use App\State\BookRepresentationProcessor;
+
+#[Post(output: AnotherRepresentation::class, processor: BookRepresentationProcessor::class)]
+class Book {}
+```
+Here the `$data` attribute represent an instance of your ressource.
```
Comment (alanpoulain) [inline core/dto.md:140]:
```suggestion

Here the `$data` attribute represents an instance of your resource.

```

File: core/dto.md [hunk 3]
```diff
@@ -119,3 +119,47 @@ final class BookRepresentationProvider implements ProviderInterface
...
+Here the `$data` attribute represent an instance of your ressource. 
+```php
+<?php
+
+namespace App\State;
+
+use App\Dto\AnotherRepresentation;
+use App\Model\Book;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
```
Comment (alanpoulain) [inline core/dto.md:149]: Please add them before.

File: core/dto.md [hunk 4]
```diff
@@ -119,3 +119,47 @@ final class BookRepresentationProvider implements ProviderInterface
...
+{
+     /**
+     * @param Book $data
+     */
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
+    {
+        return new AnotherRepresentation(
+            $data->getId(),
+            $data->getTitle(),
+            // etc
```
Comment (alanpoulain) [inline core/dto.md:161]:
```suggestion
            // etc.
```

File: core/dto.md (hunk 1 above)
Comment (Enz000) [reply core/dto.md:125]: I deleted this sentence.

File: core/dto.md (hunk 3 above)
Comment (Enz000) [reply core/dto.md:151]: In the `Provider` example, there are imported before.

Capture d’écran 2023-01-15 à 12 57 40

It's a mistake or not? We should stay coherent 😬

**Comment by Enz000** [comment]
Hey @alanpoulain 👋🏻
Thank you for your time! 
I added an xlm / YAML doc format and applied partiality your suggested changes 👍🏻

File: core/dto.md (hunk 3 above)
Comment (alanpoulain) [reply core/dto.md:178]: It's a mistake.

File: core/dto.md [hunk 5]
```diff
@@ -119,3 +119,76 @@ final class BookRepresentationProvider implements ProviderInterface
...
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- api/config/api_platform/resources.xml -->
+
+<resources xmlns="https://api-platform.com/schema/metadata/resources-3.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0
+        https://api-platform.com/schema/metadata/resources-3.0.xsd">
+    <resource class="App\Entity\Book">
+        <operations>
+            <operation class="ApiPlatform\Metadata\Post" processor="App\State\BookRepresentationProcessor"
```
Comment (alanpoulain) [inline core/dto.md:159]: Maybe add `processor` below?

Comment (Enz000) [reply core/dto.md]: done

**Review by alanpoulain — APPROVED** [review]

**Comment by alanpoulain** [comment]
Thanks @Enz000!

**Pull request merged**